src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5046,41 +5008,35 @@ internal static List<CompletionResult> CompleteVariable(CompletionContext contex
 
             if (colon == -1)
             {
-                // If no drive was specified, then look for matching drives/scopes
-                pattern = wordToComplete + "*";
-                powerShellExecutionHelper
-                    .AddCommandWithPreferenceSetting("Microsoft.PowerShell.Management\\Get-PSDrive").AddParameter("Name", pattern)
-                    .AddCommandWithPreferenceSetting("Microsoft.PowerShell.Utility\\Sort-Object").AddParameter("Property", "Name");
-                psobjs = powerShellExecutionHelper.ExecuteCurrentPowerShell(out exceptionThrown);
-                if (psobjs != null)
+                var allDrives = context.ExecutionContext.SessionState.Drive.GetAll();
```

Comment:
`Get-PSDrive` does a secrete `Import-Module` to load `Certificate` and `WSMan` drives from the `Security` module and `Microsoft.WSMan.Management` module. When calling `context.ExecutionContext.SessionState.Drive.GetAll()`, this secret auto-loading doesn't happen, so you won't get `Certificate` and `WSMan` unless those 2 modules were already loaded.

So, I think we should keep the original code for this provider completion path.

![image](https://user-images.githubusercontent.com/127450/236942068-72221d9f-b52c-4184-8325-ce58e0a2e787.png)

Reply:
Interesting. I've never really used this variable drive syntax but I've looked into it and apparently only providers that implement the IContentCmdletProvider interface are supported and neither Cert, nor WSMan implements it.  
With that in mind, I think we should stick with the current code, otherwise we would go from 100ms -> 170ms to complete.

Reply:
Great, then let's keep with the current code.